libexec/rtld-elf/rtld_c18n.c

```diff
@@ -1665,6 +1665,20 @@ tramp_reflect(const void *data)
 	return (NULL);
 }
 
+void *_rtld_tramp_reflect(const void *);
+
+void *
+_rtld_tramp_reflect(const void *addr)
+{
+	struct tramp_header *header;
+
+	header = tramp_reflect(addr);
+	if (header == NULL)
+		return (NULL);
+
+	return (header->target);
```

Comment:
Is this the actual underlying capability to the function? If so, do we not want to make it null-derived or ptraddr_t? __elf_phdr_match_addr just needs the address.

Reply:
Agreed. I can do a cheri_cleartag here.

Reply:
Why not ptraddr_t instead? It's a new API. The only reason some of the dl_* APIs use a null-derived `void *` is because the existing API uses `void *` when it really should have been something ptraddr_t-like.

Reply:
Yeah that makes sense. I'll push a patch.